provider/terraform.rb

```diff
@@ -161,5 +161,25 @@ def generate_resource_tests(data)
         out_file: filepath
       )
     end
+
+    # Only going to be run once.
+    def generate_async(output_folder, _types, version_name)
+      return if @api.objects.select(&:autogen_async).empty?
+
+      prod_name = @api.name.underscore
+      async = @api.objects.map(&:async).compact.first
```

Comment:
In Terraform, we do Operation waiting by product and in Ansible it seems you do it by resource; having to use `.first` here is a bit awkward, and makes it seem like Operation handling should either be generated separately per resource, or a product's operation should be moved from the resource level -> the product level.

Since using `.first` makes the assumption that Operations in a product are all identical, and that's our assumption based on the API, I'm strongly inclined to move the Operation object up to the product level. Most Ansible access could change from `object.async.result.resource_inside_response` to `object.__product.operation.result.resource_inside_response` to ease refactoring. What do you think?

Also just to be clear about expectations, while I'd like to know what you think we should be working towards now, and I'd like us to hit that point before being "done" generating operations, I'm fine with this solution as-is for now if we feel it makes sense as a step towards whatever our endgame is.

Reply:
I was going to add a TODO here to explain this.

I have no issues making async product-level (maybe we can override on a resource-level, but cross that bridge when we get there). I didn't want to do that on this PR because it would require a lot of api.yaml munging + Ansible changes that would clog up this PR.

Reply:
Sounds good to me! Do you mind adding the TODO or spinning out an issue so we make sure to keep track of this?